sql/drop.go

```diff
@@ -87,3 +88,72 @@ func (p *planner) DropTable(n *parser.DropTable) (planNode, error) {
 	}
 	return &valuesNode{}, nil
 }
+
+// DropDatabase drops a database.
+func (p *planner) DropDatabase(n *parser.DropDatabase) (planNode, error) {
+	if n.Name == "" {
+		return nil, errEmptyDatabaseName
+	}
+
+	nameKey := structured.MakeNameMetadataKey(structured.RootNamespaceID, string(n.Name))
+	gr, err := p.db.Get(nameKey)
+	if err != nil {
+		return nil, err
+	}
+	if !gr.Exists() {
+		if n.IfExists {
+			// Noop.
+			return &valuesNode{}, nil
+		}
+		return nil, fmt.Errorf("database %q does not exist", n.Name)
+	}
+
+	descKey := gr.ValueBytes()
+	desc := structured.DatabaseDescriptor{}
+	if err := p.db.GetProto(descKey, &desc); err != nil {
+		return nil, err
+	}
+	if err := desc.Validate(); err != nil {
+		return nil, err
+	}
+
+	tableNames, err := p.getTableNames(&desc)
+	if err != nil {
+		return nil, err
+	}
+
+	qualifiedNames := parser.QualifiedNames{}
+	for _, tableName := range tableNames {
+		qualifiedNames = append(qualifiedNames, &parser.QualifiedName{
+			Base:     parser.Name(desc.Name),
+			Indirect: parser.Indirection{parser.NameIndirection(tableName)},
+		})
+	}
+
+	if _, err := p.DropTable(&parser.DropTable{Names: qualifiedNames}); err != nil {
+		return nil, err
+	}
+
+	b := &client.Batch{}
+	b.Del(descKey)
+	b.Del(nameKey)
+	if err := p.db.Run(b); err != nil {
+		return nil, err
+	}
+	return &valuesNode{}, nil
+}
+
+func (p *planner) getTableNames(dbDesc *structured.DatabaseDescriptor) ([]string, error) {
```

Comment:
sorry, didn't see you refactored `Show()`. Either is fine with me, personally I think it would be nicer if `Show` iterated over `QualifiedNames` (which it then would make sense for `getTableNames` to return).